src/store/modules/order/actions.ts

```diff
@@ -168,12 +165,20 @@ const actions: ActionTree<OrderState, RootState> = {
           "orderId_op": "in"
         },
         "entityName": "ShipmentReceiptAndItem",
-        "fieldsToSelect": ["datetimeReceived", "productId", "quantityAccepted", "quantityRejected", "receivedByUserLoginId", "shipmentId"]
+        "fieldList": ["datetimeReceived", "productId", "quantityAccepted", "quantityRejected", "receivedByUserLoginId", "shipmentId", 'locationSeqId']
       }
       resp = await OrderService.fetchPOHistory(params)
       if (resp.status === 200 && !hasError(resp) && resp.data?.count > 0) {
         const poHistory = resp.data.docs;
         current.poHistory.items = poHistory;
+        const facilityLocationByProduct = poHistory?.reduce((products: any, item: any) => {
```

Comment:
I think we don't need optional chaining here, as we already check for resp.data.count and it that check passes then it means that we will have data in resp.data.docs